lib/portage/tests/util/file_copy/test_copyfile.py

```diff
@@ -42,25 +43,36 @@ def testCopyFileSparse(self):
             # files too big, in case the filesystem doesn't support
             # sparse files.
             with open(src_path, "wb") as f:
+                f.seek(2**16, os.SEEK_SET)
                 f.write(content)
-                f.seek(2**17, 1)
-                f.write(content)
-                f.seek(2**18, 1)
+                f.seek(2**17, os.SEEK_SET)
                 f.write(content)
                 # Test that sparse blocks are handled correctly at
-                # the end of the file (involves seek and truncate).
-                f.seek(2**17, 1)
+                # the end of the file.
+                f.truncate(2**18)
 
-            copyfile(src_path, dest_path)
+            fastcopy_success = False
+
+            def mock_fastcopy(src, dst):
+                _fastcopy(src, dst)
+                fastcopy_success = True
```

Comment:
```suggestion
                nonlocal fastcopy_success
                _fastcopy(src, dst)
                fastcopy_success = True
```

In python closures, when you want to write to a variable in the outer scope, you need to first declare it `nonlocal`

Reply:
I thought I tested this, but I think you are right. Updated.

Reply:
Retested and confirmed that nonlocal is indeed necessary here.